src/utils/convert.js

```diff
@@ -150,20 +160,43 @@ ${fields.join(',\n')}
     ]`;
 };
 
+// Get some quick metadata on auth scripting methods
+const getAuthMetaData = (definition) => {
+  const js = definition.js ? stripComments(definition.js) : '';
+
+  const hasPreOAuthTokenScripting = js.indexOf('pre_oauthv2_token') > 0;
+  const hasPostOAuthTokenScripting = js.indexOf('post_oauthv2_token') > 0;
+  const hasPreOAuthRefreshScripting = js.indexOf('pre_oauthv2_refresh') > 0;
+  const hasGetConnectionLabelScripting = js.indexOf('get_connection_label') > 0;
+
+  return {
+    hasPreOAuthTokenScripting,
+    hasPostOAuthTokenScripting,
+    hasPreOAuthRefreshScripting,
+    hasGetConnectionLabelScripting,
+  };
+};
+
+const getTestTriggerKey = (definition) => {
+  return _.get(definition, ['general', 'test_trigger_key']);
+};
+
 const renderAuthTemplate = (authType, definition) => {
-  const fields = _.map(definition.auth_fields, renderField);
+  const fields = renderFields(definition.auth_fields, 4);
   const connectionLabel = _.get(definition, ['general', 'auth_label'], '');
+  const testTriggerKey = getTestTriggerKey(definition);
+  const { hasGetConnectionLabelScripting } = getAuthMetaData(definition);
 
-  const auth = `{
-    type: '${authType}',
-    test: AuthTest.operation.perform,
-    fields: [
-${fields.join(',\n')}
-    ],
-    connectionLabel: '${connectionLabel}'
-  }`;
+  const templateContext = {
+    TEST_TRIGGER_MODULE: `./triggers/${snakeCase(testTriggerKey)}`,
+    TYPE: authType,
+    FIELDS: fields,
+    CONNECTION_LABEL: connectionLabel,
+    hasGetConnectionLabelScripting,
+  };
 
-  return Promise.resolve(auth);
+  const templateFile = path.join(TEMPLATE_DIR, '/basic-auth.template.js');
```

Comment:
We should rename this template to `simple-auth.template.js`, otherwise it seems it's dedicated for `basic` auth type only.

Reply:
Good idea! Fixed in 0cb65c9.